R/generateSettings.R

```diff
@@ -61,44 +68,29 @@ generateSettings <- function(standard="None", chart="eDish", partial=FALSE, part
     }
   }
   
-  # build shell settings for each chart 
-  # TODO: move these to `/data` eventually
-  shells<-list()
-  shells[["edish"]]<-list(
-    id_col = NULL,
-    value_col = NULL,
-    measure_col = NULL,
-    normal_col_low = NULL,
-    normal_col_high = NULL,
-    studyday_col=NULL,
-    visit_col = NULL,
-    visitn_col = NULL,
-    filters = NULL,
-    group_cols = NULL,
-    measure_values = list(ALT = NULL,
-                          AST = NULL,
-                          TB = NULL,
-                          ALP = NULL),
-    baseline = list(value_col=NULL,
-                    values=list()),
-    analysisFlag = list(value_col=NULL,
-                        values=list()),
-    
-    x_options = c("ALT", "AST", "ALP"),
-    y_options = c("TB", "ALP"),
-    visit_window = 30,
-    r_ratio_filter = TRUE,
-    r_ratio_cut = 0,
-    showTitle = TRUE,
-    warningText = "Caution: This interactive graphic is not validated. Any clinical recommendations based on this tool should be confirmed using your organizations standard operating procedures."
-  )
-  
   # loop through dataMappings and apply them to the shell
   if(standard %in% standardList){
     for(row in 1:nrow(dataMappings)){
-      shells[[chart]]<-setSettingsValue(settings = shells[[chart]], key = textKeysToList(dataMappings[row,"text_key"])[[1]], value = dataMappings[row, "column_name"])
+      shell<-setSettingsValue(settings = shell, key = textKeysToList(dataMappings[row,"text_key"])[[1]], value = dataMappings[row, "column_name"])
     }    
   }
-
-  return(shells[[chart]])
-}
\ No newline at end of file
+  
+  #############################################################################
```

Comment:
would it make sense to combine each of these loops together so that the standards and defaults are set at once?

Reply:
Yep - quite possibly. Will look in to this a bit now.

Reply:
Updated in the latest commit. Now we create a single tibble capturing text_keys and settings values for all defaults and custom values and then apply those values to the shell at the very end of `generateSettings`